crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -7185,6 +7180,13 @@ impl<'db> TypeInferenceBuilder<'db> {
                 self.infer_type_expression(slice);
                 value_ty
             }
+            Type::ClassLiteral(ClassLiteralType::Generic(generic_class)) => {
+                let specialized_class =
+                    self.infer_explicit_class_specialization(subscript, value_ty, generic_class);
+                specialized_class
+                    .in_type_expression(self.db())
+                    .unwrap_or(Type::unknown())
+            }
             _ => {
                 self.infer_type_expression(slice);
                 todo_type!("generics")
```

Comment:
This might want an update?

Reply:
Done. I got rid of the todo type completely, and instead added new diagnostics for when (a) you try to specialize a non-generic class, and (b) when you use any other subscript expression in a type expression.  Added quite a bit of churn in the mdtest suite, but I think it's worth it.